01/24/2024

0Swathi opened the pull request:
In this pull request, I took a data set called "un-general-debates-blueprint.csv" and i installed some of the packages that are pandas,matplotlib,regex ,counter and executed some of the code of sample method on dataframe,Access the column lables of a Dataframe,Acess the datatypes of  each columns in a dataframe,info()method:structure of your dataframe,describe() method:  descriptive statistics of a DataFrame's numerical columns.,Length of the speech,stats about text fields,checking the missing data,Filling the missing data,uppercase or lowercase all text and some of the charts like barplot,histogram and some code like function to tokenize,remove stop words,function to remove stop words, and we should the pipeline and run the pipeline,and we should count the tokens and we should do some wordfrequencyt,stats,and wordcloud.

0Swathi committed:
Add files via upload

nikshepkulli commented on week2_classwork_24Jan .ipynb:
```diff
@@ -0,0 +1,2159 @@
+{
```
<p>where is the result for this?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/0Swathi/txt_based_analysis/pull/5/discussion/#comment-1483389071'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='24'/>

nikshepkulli replied:
<p>legends?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/0Swathi/txt_based_analysis/pull/5/discussion/#comment-1483389072'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='28'/>

nikshepkulli replied:
<p>legends?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/0Swathi/txt_based_analysis/pull/5/discussion/#comment-1483389073'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

nikshepkulli replied:
<p>legends?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/0Swathi/txt_based_analysis/pull/5/discussion/#comment-1483389074'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

0Swathi committed:
week2

File Reading and Data Loading: It starts by importing necessary libraries and reads a CSV file containing UN general debate speeches using pandas. Then it samples 20% of the data.

Data Exploration:

Accessing column labels and data types.
Obtaining info and descriptive statistics about the DataFrame.
Calculating the length of each speech and adding it as a new column.
Checking for missing data, filling missing values, and converting text columns to uppercase.
Data Visualization:

Plotting the distribution of speech lengths using matplotlib.
Creating a box plot, histogram, and bar plot using seaborn for visualization.
Text Processing:

Tokenizing the text using regex.
Removing stopwords using NLTK.
Defining a pipeline for text processing including lowercasing, tokenization, and stopword removal.
Applying the defined pipeline to the text data.
Token Analysis:

Counting the number of tokens in each speech.
Performing word frequency analysis using Counter.
Visualizing the most common words using bar charts and word clouds.
Word Cloud Visualization: Generates a word cloud for speeches from the USA in 2015.
